build(aio): upgrade Karma and Jasmine related dependencies

gkalpak opened the pull request.

gkalpak committed:
build(aio): upgrade Karma-related dependencies

mary-poppins commented:
The angular.io preview for 3ee873814e59d3dd269ec80b4b6ee0025d2b48e8 is available [here][1].

[1]: https://pr17405-3ee873814e59d3dd269ec80b4b6ee0025d2b48e8.ngbuilds.io/

gkalpak committed:
build(aio): upgrade Jasmine-related dependencies

According to https://github.com/jasmine/jasmine/issues/1327#issuecomment-306924549,
this might help with help with frequent DISCONNECTED errors (especially on Travis).

mary-poppins commented:
The angular.io preview for 74f8e44e52494e1bab76f250cef3bb16e34a2889 is available [here][1].

[1]: https://pr17405-74f8e44e52494e1bab76f250cef3bb16e34a2889.ngbuilds.io/

gkalpak commented:
Neither commit helped with Chrome getting disconnected 😞

petebacondarwin commented:
I was just about to say that :-( I restarted the build and it still disconnected ...

gkalpak committed:
test(aio): fix Chrome getting disconnected during unit tests

Chrome (v58+) often gets disconnected during unit tests (causing them to fail).
This has been happening locally (on Windows) and on Travis. The exact reasons
are not known, but it seems that some of the `AppComponent` tests are to blame.

Based on the discussion in https://github.com/jasmine/jasmine/issues/1327 (and
plenty of trial-and-error), using Jasmine's `done()` callback before each of
test (even calling it synchronously) fixes the issue.

mary-poppins commented:
The angular.io preview for 60656632dc3fc6f2d47f4b3725325e302ffec923 is available [here][1].

[1]: https://pr17405-60656632dc3fc6f2d47f4b3725325e302ffec923.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 5c8f63cb9516d6943967bd5161410eda17cf4b2e is available [here][1].

[1]: https://pr17405-5c8f63cb9516d6943967bd5161410eda17cf4b2e.ngbuilds.io/

mary-poppins commented:
The angular.io preview for cc2d256b6e9be3700921f7a0a5e39d9ee3d2767c is available [here][1].

[1]: https://pr17405-cc2d256b6e9be3700921f7a0a5e39d9ee3d2767c.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 6263a1f4fb317b326d190f878cccf04dba221886 is available [here][1].

[1]: https://pr17405-6263a1f4fb317b326d190f878cccf04dba221886.ngbuilds.io/

mary-poppins commented:
The angular.io preview for a39bf8b3eb9e8c58c48613256a77e6aa047e88ae is available [here][1].

[1]: https://pr17405-a39bf8b3eb9e8c58c48613256a77e6aa047e88ae.ngbuilds.io/

gkalpak commented:
With this [one weird trick][1], the last 3 runs were disconnected-free 🎉 
Disconnecting was also happening consistently on my Windows machine and this fix makes it go away too (so it is not a scam 😛).

[1]: https://github.com/angular/angular/pull/17405/commits/a39bf8b3eb9e8c58c48613256a77e6aa047e88ae

petebacondarwin commented on aio/src/app/app.component.spec.ts:
```diff
@@ -59,9 +59,12 @@ describe('AppComponent', () => {
-    beforeEach(() => {
+    // For reasons unknown, not using `done` (even calling it synchronously),
+    // causes Chrome (v58+) to often get disconnected (at least on Windows and Travis).
+    beforeEach(done => {
```
would using `async` helper have the same effect or do we need to manually call `done?

gkalpak replied:
It might (e.g. I was able to fix the 4 individual tests that were causing disconnections locally, by using `async()`. IIRC using `async()` in `beforeEach` makes it more difficult/impossible to use `fakeAsync()` in a specific test.

The truth is I don't know what is causing the actual issue, so anything we do will be a work around at this stage 😞 Using `done()` seemed the most straight-forward and "lightweight" option.
